pkg/tap/switch.go

```diff
@@ -144,7 +146,10 @@ func (e *Switch) txPkt(pkt *stack.PacketBuffer) error {
 				return err
 			}
 
-			atomic.AddUint64(&e.Sent, uint64(pkt.Size()))
+			if size < 0 {
+				return fmt.Errorf("packet size out of range")
+			}
```

Comment:
You could also move this check outside of the loop right after `size := pkt.Size()`